packages/client/src/i18n/messages/constants.ts

```diff
@@ -978,6 +979,11 @@ const messagesToDefine: IConstantsMessages = {
     defaultMessage: `{registrationTargetDays} days - 1 year`,
     description: `Label for registrations within {registrationTargetDays} days to 1 year`,
     id: 'constants.withinTargetDaysTo1Year'
+  },
+  humanName: {
+    defaultMessage: `{lastName} {middleName} {firstName}`,
```

Comment:
Thanks @Zangetsu101 for this question. I was concerned about that as well, when I was manually testing this scenario yesterday it seemed to work just fine when all the `placeholders` are passed as field in the `values`, if the value is `falsey`(`null`,`false`,`undefined` except the number 0 which gets rendered as is) for a `key` it gets omitted in the rendered message.

The only time it seemed to break is when one of the  placeholders(`firstName`,`middleName,`lastName`) is missing from the values array which will result in the rendering: `{lastName} {middleName} {lastName}`

I just struggled to write tests to demostrate these scenarios because I couldn't figure out how to set different locales

Reply:
hm so you're saying even if provide an empty string for the middleName part, it renders properly? I would have guessed that it would render something like this: `John  Doe`(Notice the double space in between) for the given value: `{ firstName: "John", middleName: '', lastName: "Doe" }`

Reply:
I think a country can achieve the proper format if we default empty values as `null` than can be then matched against in the language string.

https://formatjs.github.io/docs/core-concepts/icu-syntax/

### Happy path

<img width="987" alt="image" src="https://github.com/user-attachments/assets/c8b067ec-1397-48b6-a190-8c2d15a46a4a">

### Middle name missing (extra space)
<img width="988" alt="image" src="https://github.com/user-attachments/assets/f1fa8df4-7614-4b02-b232-ea667437dd66">

### Defining empty values as `null`

```
{first}{middle, select, null {} other { {middle}}} {last}
```

<img width="1010" alt="image" src="https://github.com/user-attachments/assets/33e94604-e449-45b8-a1b7-9bc212d6ab9e">
<img width="992" alt="image" src="https://github.com/user-attachments/assets/46d0d6d3-1457-4b8e-9dfa-da0582c37489">